hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -126,54 +131,51 @@ private void removeDeadDataNodes() {
   }
 
   public List<String> getDeadJournalNodes() {
+    List<String> deadJournalHosts = new ArrayList<>();
+
     if (deadJournalNodeTimeStamp != null && deadJournalNodeTimeStamp.before(new Date())) {
       removeDeadJournalNodes();
-      return new ArrayList<>();
     } else {
       Map<String, String> journalNodes = getJournalNodes();
-      Set<String> journalHosts = journalNodes.keySet();
-      List<String> deadJournalHosts = new ArrayList<>();
-      for (String journalHost : journalHosts) {
-        if (journalNodes.get(journalHost) == null) {
-          deadJournalHosts.add(journalHost);
```

Comment:
Can you create a separate variable for journalNodes.entrySet() please? In this project, we follow the java practice to avoid too many nested method calls.